src/neps/plot/tensorboard_eval.py

```diff
@@ -0,0 +1,522 @@
+        tblogger.summary_writer.close()
+
+    @staticmethod
+    def disable(disable_logger: bool = True):
+        """
+        The function allows for enabling or disabling the logger functionality
+        throughout the program execution by updating the value of 'tblogger.disable_logging'.
+        When the logger is disabled, it will not perform any logging operations.
+
+        Args:
+            disable_logger (bool, optional): A boolean flag to control the logger.
+                                            If True (default), the logger will be disabled.
+                                            If False, the logger will be enabled.
+
+        Example:
+            # Disable the logger
+            tblogger.disable()
+
+            # Enable the logger
+            tblogger.disable(False)
+        """
+        tblogger.disable_logging = disable_logger
+
+    @staticmethod
+    def log(
+        loss: float,
+        current_epoch: int,
+        writer_scalar: bool = True,
+        writer_hparam: bool = True,
+        scalar_accuracy_mode: bool = False,
+        hparam_accuracy_mode: bool = False,
+        data: Optional[dict] = None,
+    ):
+        """
+        Log experiment data to the logger, including scalar values, hyperparameters, images, and layer gradients.
+
+        Args:
+            loss (float): The current loss value in training.
+            current_epoch (int): The current epoch of the experiment. Used as the global step.
+            writer_scalar (bool, optional): Whether to write the loss or accuracy for the
+                                        configs during training. Default is True.
+            writer_hparam (bool, optional): Whether to write hyperparameters logging
+                                        of the configs during training. Default is True.
+            scalar_accuracy_mode (bool, optional): If True, interpret the 'loss' as 'accuracy' and transform it's
+                                                value accordingliy. Default is False.
+            hparam_accuracy_mode (bool, optional): If True, interpret the 'loss' as 'accuracy' and transform it's
+                                                value accordingliy. Default is False.
+            data (dict, optional): Additional experiment data to be logged. It should be in the format:
+                                {
+                                    'tag1': tblogger.scalar_logging(value=value1),
+                                    'tag2': tblogger.image_logging(img_tensor=img, counter=2),
+                                    'tag3': tblogger.layer_gradient_logging(model=model),
+                                }
+                                Default is None.
+
+        """
+        tblogger.current_epoch = current_epoch
+        tblogger.loss = loss
+        tblogger.scalar_accuracy_mode = scalar_accuracy_mode
+        tblogger.hparam_accuracy_mode = hparam_accuracy_mode
+
+        if not tblogger.disable_logging:
```

Comment:
Another point of the double negative. Could also flatten this quite a bit with early return

```python
if not tblogger.enabled:
    tblogger.logger_bool = False
    return
    
# Logging code
```